src/algebra/linear_ordered_comm_group_with_zero.lean

```diff
@@ -52,6 +52,112 @@ instance [linear_ordered_add_comm_group_with_top α] :
+                             ... = x         : mul_one x
+                             ... ≤ 1         : H
+
+alias left.pow_le_one_of_le ← pow_le_one_of_le_one
+
+lemma left.one_le_pow_of_le : ∀ {n : ℕ} {x : α}, 1 ≤ x → 1 ≤ x^n
+| 0       x _ := (pow_zero x).symm.le
+| (n + 1) x H := calc 1 ≤ x          : H
+                    ... = x * 1      : (mul_one x).symm
+                    ... ≤ x * x ^ n  : mul_le_mul_left' (left.one_le_pow_of_le H) x
+                    ... = x ^ n.succ : (pow_succ x n).symm
+
+alias left.one_le_pow_of_le ← one_le_pow_of_one_le'
+
+lemma eq_one_of_pow_eq_one {α : Type*} [monoid α] [linear_order α]
+  [covariant_class α α (*) (≤)]
+  [covariant_class α α (function.swap (*)) (≤)]
+  {x : α} {n : ℕ} (hn : n ≠ 0) (H : x ^ n = 1) : x = 1 :=
+begin
+  rcases nat.exists_eq_succ_of_ne_zero hn with ⟨n, rfl⟩, clear hn,
+  induction n with n ih,
+  { exact (pow_one _).symm.trans H },
+  { cases le_total x 1 with h,
+    all_goals
+    { have h1 := mul_le_mul_right' h (x ^ (n + 1)),
+      rw pow_succ at H,
+      rw [H, one_mul] at h1 },
+    { have h2 := pow_le_one_of_le_one h,
+      exact ih (le_antisymm h2 h1) },
+    { have h2 := one_le_pow_of_one_le' h,
+      exact ih (le_antisymm h1 h2) } }
+end
+
+end left
+
+section right
+variable [covariant_class α α (function.swap (*)) (≤)]
+lemma right.one_le_pow_of_le {x : α} (H : 1 ≤ x) :
+  ∀ {n : ℕ}, 1 ≤ x^n
+| 0       := (pow_zero _).symm.le
+| (n + 1) := calc 1 ≤ x          : H
+                ... = 1 * x      : (one_mul x).symm
+                ... ≤ x ^ n * x  : mul_le_mul_right' right.one_le_pow_of_le x
+                ... = x ^ n.succ : (pow_succ' x n).symm
+
+lemma right.pow_le_one_of_le {x : α} (H : x ≤ 1) :
+  ∀ {n : ℕ}, x^n ≤ 1
+| 0       := (pow_zero _).le
+| (n + 1) := calc x ^ n.succ = x ^ n * x : pow_succ' x n
+                         ... ≤ 1 * x     : mul_le_mul_right' right.pow_le_one_of_le x
+                         ... = x         : one_mul x
+                         ... ≤ 1         : H
+
+end right
+
+lemma pow_le_pow_of_le [covariant_class α α (*) (≤)] [covariant_class α α (function.swap (*)) (≤)]
+  {x y : α} (H : x ≤ y) :
+  ∀ {n : ℕ} , x^n ≤ y^n
+| 0       := (pow_zero _).le.trans (pow_zero _).symm.le
+| (n + 1) := calc  x ^ n.succ = x * x ^ n  : pow_succ x n
+                          ... ≤ y * x ^ n  : mul_le_mul_right' H (x ^ n)
+                          ... ≤ y * y ^ n  : mul_le_mul_left' pow_le_pow_of_le y
+                          ... = y ^ n.succ : (pow_succ y n).symm
+
+@[elab_as_eliminator]
+lemma induction_from_zero_lt {p : ℕ → Prop} {n : ℕ} (n0 : 0 < n)
+  (p1 : p 1) (pind : ∀ {n}, 0 < n → p n → p n.succ) :
+  p n :=
+begin
+  rcases nat.exists_eq_succ_of_ne_zero n0.ne' with ⟨n, rfl⟩,
+  induction n with n hn,
+  { exact p1 },
+  { exact pind n.succ_pos (hn n.succ_pos) }
+end
+
+lemma left.pow_lt_one_of_lt [covariant_class α α (*) (<)] {n : ℕ} {x : α} (n0 : 0 < n) (H : x < 1) :
+  x^n < 1 :=
```

Comment:
Can you add `iff` versions of this and the next lemma under stricter assumptions?